bitrise/util.go

```diff
@@ -245,7 +245,58 @@ func ReadSpecStep(pth string) (stepmanModels.StepModel, error) {
 	return stepModel, nil
 }
 
-func fillStepOutputs(stepListItem *models.StepListItemModel, defaultStepLibSource string) error {
+func getInputByKey(inputs []envmanModels.EnvironmentItemModel, key string) (envmanModels.EnvironmentItemModel, error) {
+	for _, input := range inputs {
+		aKey, _, err := input.GetKeyValuePair()
+		if err != nil {
+			return envmanModels.EnvironmentItemModel{}, err
+		}
+		if aKey == key {
+			return input, nil
+		}
+	}
+	return envmanModels.EnvironmentItemModel{}, fmt.Errorf("No Environmnet found for key (%s)", key)
+}
+
+func isStringSliceWithSameElements(s1, s2 []string) bool {
+	if len(s1) != len(s2) {
+		return false
+	}
+
+	m := make(map[string]bool, len(s1))
+	for _, s := range s1 {
+		m[s] = true
+	}
+
+	for _, s := range s2 {
+		v, found := m[s]
+		if !found || !v {
+			return false
+		}
+	}
+	return true
+}
+
+func isDependencieSliceWithSameElements(s1, s2 []stepmanModels.DependencyModel) bool {
+	if len(s1) != len(s2) {
+		return false
+	}
+
+	for _, dep1 := range s1 {
+		for i, dep2 := range s2 {
+			if dep1.Manager == dep2.Manager && dep1.Name == dep2.Name {
+				continue
+			}
+			if i == len(s2)-1 {
+				return false
+			}
+		}
+	}
+
+	return true
+}
+
+func removeStepDefaultsAndfillStepOutputs(stepListItem *models.StepListItemModel, defaultStepLibSource string) error {
```

Comment:
func name type - `fill` should be `Fill` : `removeStepDefaultsAndFillStepOutputs`